### ISSUE TITLE
Allow polyline5 input/output

### ISSUE THREAD

**Issue by dnesbitt61** [valhalla/valhalla#1686]
Code was added to perform polyline5 encoding and decoding as well as a shape_format option. Currently this only applies to the shape output when OSRM compatible output is requested. With the underyling polyline5 support, Valhalla could be updated to use polyline5  for Valhalla JSON responses as well as allow polyline5 as input for map matching and elevation request shape input. 

To do this we need to:

1. update code for parsing encoded polyline inputs
2. update docs
3. possibly update decoding sample code (currently uses polyline6 which is the Valhalla default)

**Comment by mqln** [comment]
@dnesbitt61 Is it currently possible toggle the `shape_format` `DirectionsOption` while using the `format: osrm DirectionsOption`, or does it always override to polyline5? The current output with `format: osrm` works great, except for some older clients who're on an older version of Mapbox, which I'm fairly sure is still dependent on polyline6. In otherwords, the actual route points are getting decoded in theses clients as one decimal point off (i.e. 40.000 -90.000 is 400.000, -900.000)

**Comment by dnesbitt61** [comment]
you should be able to set shape_format - polyline5 in the JSON request and use that in conjunction with OSRM format. That is what the flag was added for...do you have an example where it does not work? The default should be polyline6.

**Comment by mqln** [comment]
Yeah, here's a sample curl, and please excuse me if I'm not formatting something correctly.

`'{"locations":[{"lat":44.9778,"lon":-93.2650,"type":"break"},{"lat":37.7749,"lon":-122.4194,"type":"break"}],"costing":"auto","directions_options":{''format":"osrm","shape_format":"polyline5"}}'`

**Comment by dnesbitt61** [comment]
looks like shape_format needs to be outside the directions_options:
{"locations": [{"lat": 44.9778,"lon": -93.2650,"type": "break"}, {"lat": 37.7749,"lon": -122.4194,"type": "break"}],"costing": "auto","directions_options": {"format": "osrm"},"shape_format": "polyline6"}
or
{"locations": [{"lat": 44.9778,"lon": -93.2650,"type": "break"}, {"lat": 37.7749,"lon": -122.4194,"type": "break"}],"costing": "auto","directions_options": {"format": "osrm"},"shape_format": "polyline5"}

**Comment by dnesbitt61** [comment]
Seems like we should make these consistent - so I will leave this issue open and hopefully resolve shortly. these are undocumented features, so they might change.

**Comment by mqln** [comment]
Ahh beautiful, thanks for resolving that. Cheers.

**Comment by drewda** [comment]
@dnesbitt61 is the goal here to allow polyline5 input/output as an option behind a param, or to switch the default input/output to polyline5? The former sounds great. The latter could be a major issue for @interline-io's client libraries and users.

**Comment by dnesbitt61** [comment]
Currently this is just for the OSRM compatible output. The intention is to leave polyline6 the default and PERHAPS document the option and allow polyline5 output for Valhalla. I hesitate a bit on this as I think that offering polyline5 as an input is risky. I fear that edge walking would not work well for trace_attributes so my instinct is to only allow polyline6 as input.

**Comment by kevinkreiser** [comment]
I second @dnesbitt61 's concern about edge walking with polyline5, very good point!

**Comment by drewda** [comment]
Good to know. Thanks, @dnesbitt61

**Comment by danpat** [comment]
And just to chime in on why OSRM has polyline5 at all - it's the default for the Google (and other) polyline decoding libraries, so we made it the default to silence all the support tickets we were getting when people were decoding our (previously) polyline6 strings incorrectly :-P

**Comment by kazuhidelee** [comment]
@dnesbitt61  
Hi I was wondering if this issue is still relevant? If yes, I would love to work on this :)

**Comment by chrstnbwnkl** [comment]
This one can actually be closed, `polyline5` was introduced ages ago.
